src/lua/LuaFileSystem.cpp

```diff
@@ -7,15 +7,107 @@
 #include "LuaObject.h"
 #include "Pi.h"
 
-/*
- * Interface: FileSystem
- *
- * A global table that provides access to the filesystem.
- *
- * This interface is protected. It can only be used by scripts in Pioneer's
- * data directory. Mods and scripts in the user directory that try to use it
- * will get a Lua error.
- */
+#include "core/StringUtils.h"
+
+struct SplitURI
+{
+	std::string		 full_path;
+	std::string_view scheme;
+	std::string_view rel_path;
+};
+
+SplitURI ParseURIArg(lua_State* l, int numArg)
+{
+	SplitURI res;
+	size_t path_size;
+	const char* path_str = luaL_checklstring(l, numArg, &path_size);
```

Comment:
As long as the underlying string value isn't removed from the stack, this pointer will remain valid. You can store `full_path` as a `string_view` (initialized via ptr + size constructor) as long as the methods using SplitURI don't pop the value from the stack.

Reply:
Ahh, for some reason, I thought `std::string_view` didn't have the `find_first_of` method, which took me down the wrong path - thanks :-)

Reply:
I also just realized that `find_first_of` is I believe incorrect here, we should be using `find` as we want to match the entire `://` string, not any char in it.

Reply:
Indeed, `string_view::find()` is probably what we want here - good catch!